pinot-core/src/main/java/org/apache/pinot/core/routing/RoutingManager.java

```diff
@@ -117,4 +117,11 @@ default boolean isTableDisabled(String tableNameWithType) {
    */
   @Nullable
   Set<String> getServingInstances(String tableNameWithType);
+
+  /**
+   * Returns the query timeout in milliseconds for the given table.
+   * Returns {@code null} if the table does not have a query timeout set.
+   */
+  @Nullable
+  Long getQueryTimeoutMs(String tableNameWithType);
```

Comment:
query timeout should be associated with something like a `RequestContext`. Can we avoid this method?